scipy/stats/tests/test_mstats_basic.py

```diff
@@ -230,6 +230,16 @@ def test_spearmanr(self):
               0.0, 0.6, 6.7, 3.8, 1.0, 1.2, 1.4, np.nan]
         (x, y) = (ma.fix_invalid(x), ma.fix_invalid(y))
         assert_almost_equal(mstats.spearmanr(x,y)[0], 0.6887299)
+
+        # test alternative parameter (based on above)
+        pr = mstats.spearmanr(x, y).pvalue
+
+        p = mstats.spearmanr(x, y, alternative="greater").pvalue
+        assert_almost_equal(p, pr / 2)
```

Comment:
I think this is right, but how did you know for this data which was the correct value for each alternative?

Reply:
The way I look at it is that the statistic (expected[0]) is greater than 0 in this case, so we expect the pre-multiplied p-value to have been calculated with the SF, like alternative='greater' is. Therefore, the expected p-value only needs to be divided by two to equal the "greater" alternative, while the "less" alternative must be reversed by subtracting from 1 before reversal, because it is found using the CDF.

Reply:
The explanation of the alternatives is not explicit about what "greater" or "less" mean in the context of the test. Most tests for which there is already an `alternative` keyword are no better (e.g. [mannwhitneyu](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.mannwhitneyu.html#scipy.stats.mannwhitneyu) gives zero information about what the test compares, [brunnermunzel](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.brunnermunzel.html#scipy.stats.brunnermunzel) is not much more clear about the alternatives), so I don't fault this PR for that; I think after we get the code in there, a round of documentation improvement is needed. But we need to be careful here, because it is not always obvious what the alternative hypotheses mean, and sometimes it is easy (e.g. gh-13449) to get them exactly reversed. So I want to make sure that we have, at least internally, a clear understanding of what the alternatives mean in each case. I think your logic - basically, "greater" uses the survival function and "less" deals with the CDF, so if the test statistic is greater than the median and the alternative is "greater", the p-value should be less than 1/2 - will usually be correct. But I think we need to be more careful than that. 

Actually... I think I'm changing my mind; we really should not follow precedent, and we should document what the alternatives mean very explicitly for these new PRs. Can you research these tests and spell out what these alternatives mean in the documentation? The labels "greater" and "less" are, to some extent, a convention (e.g. is A _greater than_ B, or is B _greater than_ A), so what convention is used by other languages for these routines?

Sorry to make the scope explode, but after seeing gh-13449 I think it's important. Let's split do future `alternative` PRs one function at a time to keep them more focused.

Reply:
Meh, don't worry about that for this PR. We will go back later. For this one, I think just address the comments in the code and we'll be set.